hail/src/main/scala/is/hail/expr/ir/lowering/LowerTableIR.scala

```diff
@@ -820,12 +820,19 @@ object LowerTableIR {
 
         val sortFields = newKeyType.fieldNames.map(fieldName => SortField(fieldName, Ascending)).toIndexedSeq
         val withNewKeyRType = TypeWithRequiredness(shuffledRowType).asInstanceOf[RStruct]
+        analyses.requirednessAnalysis.lookup(newKey).asInstanceOf[RStruct]
+          .fields
+          .foreach { f =>
+            withNewKeyRType.field(f.name).unionFrom(f.typ)
+          }
+
         val shuffled = ctx.backend.lowerDistributedSort(
           ctx, partiallyAggregated , sortFields, relationalLetsAbove, withNewKeyRType)
         val repartitioned = shuffled.repartitionNoShuffle(shuffled.partitioner.strictify)
 
-        val takeAggSig = PhysicalAggSig(Take(), TakeStateSig(VirtualTypeWithReq(shuffledRowType ,TypeWithRequiredness(shuffledRowType))))
-        val aggStateSigsPlusTake = aggs.states ++ Array(TakeStateSig(VirtualTypeWithReq(shuffledRowType ,TypeWithRequiredness(shuffledRowType))))
+        val takeVirtualSig = TakeStateSig(VirtualTypeWithReq(shuffledRowType ,TypeWithRequiredness(shuffledRowType)))
+        val takeAggSig = PhysicalAggSig(Take(), takeVirtualSig)
+        val aggStateSigsPlusTake = aggs.states ++ Array(takeVirtualSig)
```

Comment:
Also, this looks unused.

Reply:
used on L872

Reply:
Oh sorry, misread the diff and thought you added it on the last commit.